ui/apps/platform/cypress/integration/certexpiration.test.js

```diff
@@ -1,125 +1,175 @@
-            cy.intercept('GET', api.permissions.mypermissions, {
-                body: {
-                    globalAccess: 'READ_ACCESS',
-                    resourceToAccess: {
-                        VulnerabilityManagementRequests: 'READ_ACCESS',
-                        VulnerabilityManagementApprovals: 'READ_ACCESS',
-                    },
-                },
-            }).as('permissions');
             const expiry = dateFns.addMinutes(dateFns.addHours(new Date(), 23), 30);
-            mockCertExpiryAndVisitHomepage(api.certExpiry.central, expiry);
-            cy.wait('@permissions');
+            const staticResponseMapGeneric = {
+                [mypermissionsAlias]: {
+                    fixture: 'auth/mypermissionsMinimalAccess.json',
+                },
+                [credentialexpiryCentralAlias]: {
+                    body: { expiry },
+                },
+            };
+            visitForCredentialExpiry(staticResponseMapGeneric);
+
             cy.get(selectors.centralCertExpiryBanner)
                 .invoke('text')
                 .then((text) => {
                     expect(text).to.include('Central certificate expires in 23 hours');
                     expect(text).to.include('Contact your administrator');
                 });
-            cy.get(selectors.centralCertExpiryBanner).find('button').should('not.exist');
+            cy.get(`${selectors.centralCertExpiryBanner} button`).should('not.exist');
         });
 
         it('should show a warning banner if the expiry date is within 4-14 days', () => {
             const expiry = dateFns.addDays(new Date(), 10);
-            mockCertExpiryAndVisitHomepage(api.certExpiry.central, expiry);
+            const staticResponseMapGeneric = {
+                [credentialexpiryCentralAlias]: {
+                    body: { expiry },
+                },
+            };
+            visitForCredentialExpiry(staticResponseMapGeneric);
 
             cy.get(selectors.centralCertExpiryBanner).should('have.class', 'pf-m-warning');
         });
 
         it('should show a danger banner if the expiry date is less than or equal to 3 days', () => {
             const expiry = dateFns.addDays(new Date(), 2);
-            mockCertExpiryAndVisitHomepage(api.certExpiry.central, expiry);
+            const staticResponseMapGeneric = {
+                [credentialexpiryCentralAlias]: {
+                    body: { expiry },
+                },
+            };
+            visitForCredentialExpiry(staticResponseMapGeneric);
 
             cy.get(selectors.centralCertExpiryBanner).should('have.class', 'pf-m-danger');
         });
 
         it('should download the YAML', () => {
             const expiry = dateFns.addDays(new Date(), 1);
-            mockCertExpiryAndVisitHomepage(api.certExpiry.central, expiry);
+            const staticResponseMapGeneric = {
+                [credentialexpiryCentralAlias]: {
```

Comment:
I was wondering about the possibility of defining a reusable object like:
```
const centralCredentialExpiryResponse = {
    [credentialexpiryCentralAlias]: {
        body: { expiry },
    }
}

const mypermissionsResponse = {
    [mypermissionsAlias]: {
        fixture: 'auth/mypermissionsMinimalAccess.json',
    }
}
```
Then we can reuse blocks of these for the `staticResponseMapGeneric` objects
```
const staticResponseMapGeneric = {
    ...centralCredentialExpiryResponse,
    ...mypermissionsResponse
}
```